#!/usr/bin/env python3
PACKAGE = "spinnaker_camera_driver_helpers"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_framerate",    int_t,    0, "Framerate limiter for cameras", 4,  0, 32)
gen.add("exposure",    int_t,    0, "Exposure time for cameras", 4000,  0, 20000)
gen.add("gamma",    double_t,    0, "Gamma", 0.6,  0, 2)

gen.add("balance_ratio",    double_t,    0, "Balance ratio", 0,  0, 4)
gen.add("gain",    int_t,    0, "Gain", 0,  0, 27)
gen.add("grey_value",    int_t,    0, "Target grey value", 0,  0, 100)


exit(gen.generate(PACKAGE, "spinnaker_camera_driver_helpers", "CameraArraySettings"))
